forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -484,7 +484,7 @@ pub async fn deploy_executables(
         if let Some(loader_data_section_offset) =
             loader_data_offset(&pkg.bytecode.bytes, &BlobId::default())?
         {
-            if let ProgramABI::Fuel(mut fuel_abi) = pkg.program_abi.clone() {
+            if let ProgramABI::Fuel(mut fuel_abi) = pkg.program_abi.as_ref().clone() {
```

Comment:
Just checking... couldn't this be left as `pkg.program_abi.clone()`?

Reply:
Does not compile without `.as_ref()`; the behaviour remains the same; the inner data is still copied over (as per original implementation).
A larger refactor would be required to actually utilize the arc count :/

Reply:
Unresolving this to make ensure you see this one, if you still feel like this is ok feel free to resolve:

I think this makes this a little bit anti-pattern as a reader of this code i see that an Arc is introduced my initial thought is that we do this remove deep copies of abi.  So while reviewing i thouhgt even though it is not related tot he feature implemented here it is ok not to rever this. But this can be dangerous in the later run if we end-up needing to profile things. We had instances where big clones hurt the memory usage heavily and this change might make the initial diagnosis for something like that a little hard to detect.

Reply:
I think the `Arc` is quite helpful to have in the struct for any downstream references; there are more upside(s) of keeping it there - rather than having to re-introduce it if anything else downstream needs to clone it.
Instead i'll create a gh issue and reference it in the code.

Reply:
addressed: https://github.com/FuelLabs/sway/pull/6996/commits/42b6e67d3c02352a9fc8bd0749048e3b8979e723
issue: https://github.com/FuelLabs/sway/issues/6997